### DISCUSSION TITLE
Workspace Tags

### DISCUSSION THREAD

**Discussion by atagen** [YaLTeR/niri#449] [Ideas]
This isn't really a hard request or even firm idea, more of just a light probing to see how people feel about an option to use Tags.

Effectively what I envision is the standard-ish tag setup, a single workspace where windows are assigned to a tag, then tags are de/activated to make all windows within that tag visible or not.

I believe tagging columns to re/disappear them would be particularly smooth in Niri, taking the vertical constraint as its advantage to make this feel much more intuitive than an autotiler.

I think this would contribute to keeping users in flowstate much easier; I personally find full workspace transitions quite disruptive, and I tend to subconciously use whatever means a compositor has - stacking, scratchpads, etc - to avoid them.

**Comment by YaLTeR** [comment]
What if you end up with one window visible on two workspaces on two differently sized monitors at once? Wouldn't that mean the window has to have two different sizes at once?

**Comment by atagen** [comment]
I would expect the tag to have affinity with whichever workspace it was activated in last.

So to be clear, it would unmap the tag from the first workspace anyway, effectively moving it across monitors.

**Reply by YaLTeR** [reply]
Not sure I understand what you mean by that?

**Reply by atagen** [reply]
Sorry, just edited it to clarify as you posted that

**Reply by YaLTeR** [reply]
Hm, that sounds like quite a bit different of a system compared to current. I guess if at some point in the future we'll be adding other workspace systems (static workspaces come to mind) we can consider this, but for now I've got other things to worry about than different workspace systems :sweat_smile:
